clients/src/main/java/org/apache/kafka/clients/MetadataCache.java

```diff
@@ -39,8 +39,9 @@
 import java.util.stream.Collectors;
 
 /**
- * An internal mutable cache of nodes, topics, and partitions in the Kafka cluster. This keeps an up-to-date Cluster
+ * An internal immutable cache of nodes, topics, and partitions in the Kafka cluster. This keeps an up-to-date Cluster
  * instance which is optimized for read access.
+ * Prefer to extend MetadataCache's API for internal client usage Vs the public {@link Cluster}
  */
 public class MetadataCache {
```

Comment:
We don't have to do it here, but "snapshot" might be a better name since it suggests immutability.

Reply:
done